theano/gpuarray/c_code/dnn_redux.c

```diff
@@ -97,6 +97,49 @@ int APPLY_SPECIFIC(dnn_redux)(PyGpuArrayObject *input,
       PyErr_Format(PyExc_RuntimeError, "GpuArray_reshape_inplace: %s", GpuArray_error(&(*output)->ga, err));
       return 1;
     }
+
+    if (rsz == 1 && cudnnGetVersion() <= 7004) {
```

Comment:
I think this is dangerous. We don't know when this support will be added in cudnn. So for new version of cudnn, we should do the work around I think. We should not keep the old behavior of returning wrong result. Only when we tested a new version that don't request this, we can remove the work around. (Or return an error for newer cudnn version, but I don't like this option)